### PR TITLE
Fix overflow check for int64 tensors in pad_sequence

### LINKED ISSUE #159363: pad_sequence overflows when padding with larger value than int64.max - 512
### 🐛 Describe the bug

pad_sequence fails if padding int64 tensor with a padding value larger than `torch.iinfo(torch.int64).max - 511`

```
import torch

x=torch.nn.utils.rnn.pad_sequence(
                [torch.tensor([1,2,3],dtype=torch.int64),  torch.tensor([1,2],dtype=torch.int64)], padding_value=torch.iinfo(torch.int64).max - 511
            )
print(x)
```

output
```
tensor([[                   1,                    1],
        [                   2,                    2],
        [                   3, -9223372036854775808]])
```

### Versions

```
Collecting environment information...
PyTorch version: 2.8.0.dev20250529+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.2 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: 18.1.8 (++20240731025043+3b5b5c1ec4a3-1~exp1~20240731145144.92)
CMake version: version 3.31.5
Libc version: glibc-2.39

Python version: 3.12.3 (main, Jun 18 2025, 17:59:45) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-64-generic-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 4070
Nvidia driver version: 570.133.20
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               448
On-line CPU(s) list:                  0-447
Vendor ID:                            AuthenticAMD
Model name:                           AMD EPYC 9734 112-Core Processor
CPU family:                           25
Model:                                160
Thread(s) per core:                   2
Core(s) per socket:                   112
Socket(s):                            2
Stepping:                             2
Frequency boost:                      enabled
CPU(s) scaling MHz:                   78%
CPU max MHz:                          2200.0000
CPU min MHz:                          1500.0000
BogoMIPS:                             4399.57
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local user_shstk avx512_bf16 clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin cppc amd_ibpb_ret arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif x2avic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid overflow_recov succor smca fsrm flush_l1d debug_swap
Virtualization:                       AMD-V
L1d cache:                            7 MiB (224 instances)
L1i cache:                            7 MiB (224 instances)
L2 cache:                             224 MiB (224 instances)
L3 cache:                             512 MiB (32 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-111,224-335
NUMA node1 CPU(s):                    112-223,336-447
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Mitigation; Safe RET
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] Could not collect
[conda] Could not collect
```

### PR DESCRIPTION
**Pull request by ghostspiders**
Fixes #159363
## Summary
Added overflow validation in `pad_sequence` function to prevent integer overflow when padding sequences with `int64` tensors.

## Changes
- Added overflow check in `aten/src/ATen/native/PackedSequence.cpp`
- Validates `padding_value` against `static_cast<int64_t>(padding_value)` 
- Throws `RuntimeError` with message "cannot be exactly represented as int64_t" if overflow detected

## Testing
- Added test case to verify overflow detection works correctly
- Existing tests continue to pass
cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

### CODE REVIEW

**Comment by ghostspiders** [comment]
@albanD Could you please review this PR when you have time? Thanks for your time 。

**Review by ezyang — CHANGES_REQUESTED** [review]
This needs to be implemented more carefully. Please reflect on potential problems and then improve it.

**Comment by ghostspiders** [comment]
@ezyang  Thanks for the review! I've revised the validation approach.

**Question:** Would `TORCH_WARN` be safer than `TORCH_CHECK` here?

**Review by ezyang — APPROVED** [review]
This is a good. Better would be to turn padding into a Scalar so that we can represent this exactly but that is going to be a BC problem and I'm not sure how easily we can do this (maybe an extra overload will work).

**Review by ezyang — CHANGES_REQUESTED** [review]
No, the new code I don't think is right anymore
